pulsefire/ratelimiters.py

```diff
@@ -91,12 +94,13 @@ async def acquire(self, invocation: Invocation) -> float:
             ("method", 1, invocation.params.get("region", ""), invocation.method, invocation.urlformat),
         ]:
             count, limit, expire, latency, pinged = self._index[target]
+            adjusted_limit = int(limit * self.limiting_share)
```

Comment:
I prefer the limit adjustment to happen on `synchronize` instead of doing a float multiplication and then an integer casting per request; given the volume of requests it will handle, this could become inefficient.
Look for the line `self._index[(scope, idx, *subscopes)] = (` in the `synchronize` method, that is where it sets the limit.